DataStructures.Tests/AATreeTests.cs

```diff
@@ -0,0 +1,221 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using DataStructures.AATree;
+
+using NUnit.Framework;
+
+namespace DataStructures.Tests
+{
+    class AATreeTests
+    {
+        [Test]
+        public static void Constructor_UseCustomComparer_FormsCorrectTree()
+        {
+            var tree = new AATree<int>(Comparer<int>.Create((x, y) => x.CompareTo(y) * -1));
+            tree.AddRange(new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 });
+            Assert.AreEqual(1, tree.GetMax());
+            Assert.AreEqual(10, tree.GetMin());
+
+            var expected = new int[] { 10, 9, 8, 7, 6, 5, 4, 3, 2, 1 };
+            var actual = tree.GetKeysInOrder();
+            Assert.IsTrue(Enumerable.SequenceEqual(expected, actual));
+
+            Assert.DoesNotThrow(() => tree.Validate());
+        }
+
+        [Test]
+        public static void Add_MultipleKeys_FormsCorrectTree()
+        {
+            var tree = new AATree<int>();
+
+            foreach(int elem in new int[] { 1, 2, 3, 4, 5, 6, 7, 8, 9 ,10 })
```

Comment:
Codacy found an issue: [Remove the array type; it is redundant.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6367315)